### ISSUE TITLE
Leverage configelement.module for a domain_config_ui

### ISSUE THREAD

**Issue by axel-rutz** [agentrickard/domain#452]
I made a step towards autogenerating config forms from schema by creating https://www.drupal.org/project/configelement.

In fact it is an generalization of what language overrides do for their use case, so i think it can be of use here. I won't be ablle to put much energy into this, but feel free to ping me if some configelement adjustments need go in.

**Comment by agentrickard** [comment]
How did you overcome the inability to discover all config objects and their routes?

**Comment by axel-rutz** [comment]
I would not do that as core does not provide an API and this would be fighting against windmills.

What we can do though is like translation does, provide a central browser for **all** config to vary by domain. And like translation does, if a module provides the config path info, then we can use it. I think we should be able to leverage the translation hook-or-was it-yml-file for that.

**Comment by agentrickard** [comment]
No, what I'm saying is that you *cannot* do what core language does because it is treated like a unique snowflake thanks to how `*.config_translation.yml` files work to map routes to config.

And Config Translation *only* cares about strings, so it is useless to us as a model.

Trust me, I've tried. There is an issue on d.o. about this somewhere.

See the approach in #434 which actually works quite well by checking if the form has a config option.

**Comment by agentrickard** [comment]
Here's the core issue that lays out the problem.

https://www.drupal.org/project/drupal/issues/2677124

Right now, I'm following the path outlined in #434, which is about as close as we can get.

**Comment by agentrickard** [comment]
So if I understand correctly, ConfigElement creates a new form field and the idea is that we form_alter() it in to configuration we want to save? It's not immediately clear.
